src/buildercore/utils.py

```diff
@@ -308,3 +309,33 @@ def hasallkeys(ddict, key_list):
 def missingkeys(ddict, key_list):
     "returns all keys in key_list that are not in given ddict"
     return [key for key in key_list if key not in ddict]
+
+
+def fab_get(remote_path, local_path=None, use_sudo=False, label=None, return_stream=False):
+    "wrapper around fabric.operations.get"
+    label = label or remote_path
+    msg = "downloading %s" % label
+    LOG.info(msg)
+    local_path = local_path or BytesIO()
+    get(remote_path, local_path, use_sudo=use_sudo)
+    if isinstance(local_path, BytesIO):
+        if return_stream:
+            local_path.seek(0) # reset stream's internal pointer
+            return local_path
+        return local_path.getvalue().decode() # return a string
+    return local_path
+
+def fab_put(local_path, remote_path, use_sudo=False, label=None):
+    "wrapper around fabric.operations.put"
+    label = label or local_path
+    msg = "uploading %s to %s" % (label, remote_path)
+    LOG.info(msg)
+    put(local_path=local_path, remote_path=remote_path, use_sudo=use_sudo)
+    return remote_path
+
+def fab_put_data(data, remote_path, use_sudo=False):
+    ensure(isinstance(data, bytes) or isinstance(data, str), "data must be bytes or a string that can be encoded to bytes")
+    data = data if isinstance(data, bytes) else data.encode()
```

Comment:
maybe specify `utf-8` as an argument to `encode()` to be unambiguous?

Reply:
I think as we get more familiar with python3 the `.encode` and `.decode` calls will become very familiar.